pkg/tsdb/azuremonitor/azuremonitor.go

```diff
@@ -55,7 +55,8 @@ func ProvideService(cfg *setting.Cfg, httpClientProvider *httpclient.Provider, r
 		CallResourceHandler: httpadapter.New(resourceMux),
 	})
 
-	if err := registrar.LoadAndRegister(dsName, factory); err != nil {
+	resolver := plugins.CoreDataSourcePathResolver(cfg, pluginID)
+	if err := pluginStore.AddWithFactory(context.Background(), pluginID, factory, resolver); err != nil {
```

Comment:
Nit/something we can do in a separate PR. I would like to remove this code from all core plugins. Create a `CorePluginsService` or similar that takes all the core plugin services as dependencies and register them all. Then we only need to make changes to the CorePluginsService for all core plugins when needed instead of each one. How it's being registered should be a responsibility for someone else. WDYT?

Reply:
Yep agree. Was trying to find a nice pattern before, but couldn't land on anything. Now I think things are in better shape